collector/lib/runtime-control/Config.cpp

```diff
@@ -0,0 +1,130 @@
+#include <runtime-control/Config.h>
+
+#include "Logging.h"
+
+namespace collector::runtime_control {
+
+Config& Config::GetOrCreate() {
+  static Config config;
+
+  return config;
+}
+
+bool Config::WaitUntilInitialized(unsigned int timeout_ms) {
+  std::unique_lock<std::mutex> lock(mutex_);
+
+  return config_message_ || (condition_.wait_for(lock, std::chrono::milliseconds(timeout_ms)) == std::cv_status::no_timeout);
+}
+
+void Config::Update(const storage::RuntimeFilteringConfiguration& msg) {
+  std::unique_lock<std::mutex> lock(mutex_);
+
+  config_message_.emplace(msg);
+  ConfigMessageToConfig(msg);
+  std::string cluster = "cluster-1";  // Cluster is always the same, so should probably not use it anywhere. Using it but hardcoding it for now.
+  SetBitMasksForNamespaces(cluster);
+
+  condition_.notify_all();
+}
+
+bool Config::IsFeatureEnabled(uint64_t bitMask, storage::RuntimeFilterFeatures feature) {
+  return (bitMask & kFeatureFlags_[feature]) != 0;
+}
+
+void Config::ConfigMessageToConfig(const storage::RuntimeFilteringConfiguration& msg) {
+  for (storage::RuntimeFilter runtimeConfig : msg.runtime_filters()) {
+    CLOG(INFO) << "runtimeConfig.feature()= " << runtimeConfig.feature();
+    if (runtimeConfig.default_status().compare("on") == 0) {
+      default_status_map_[runtimeConfig.feature()] = true;
+      CLOG(INFO) << "Setting default to true";
+    } else {
+      default_status_map_[runtimeConfig.feature()] = false;
+      CLOG(INFO) << "Setting default to false";
+    }
+
+    for (auto runtimeConfigRule : runtimeConfig.rules()) {
+      config_by_feature_[runtimeConfig.feature()].push_back(runtimeConfigRule);
+      CLOG(INFO) << "Adding rule to config";
+    }
+  }
+
+  for (storage::ResourceCollection rc : msg.resource_collections()) {
+    rcMap_[rc.id()] = rc;
+    CLOG(INFO) << "Adding rc.id()= " << rc.id();
+  }
+}
+
+bool Config::IsFeatureEnabled(std::string cluster, std::string ns, storage::RuntimeFilterFeatures feature) {
+  std::vector<storage::RuntimeFilter_RuntimeFilterRule> filteringRules;
+  bool defaultStatus = false;
+  auto filteringRulesPair = config_by_feature_.find(feature);
```

Comment:
```suggestion
  const auto& filteringRulesPair = config_by_feature_.find(feature);
```

Reply:
config_by_feature is now an array instead of a map so this no longer applies.